sled-agent/src/params.rs

```diff
@@ -843,9 +844,25 @@ impl TryFrom<ServiceZoneService>
 /// Used to request that the Sled initialize multiple services.
 #[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq)]
 pub struct ServiceEnsureBody {
+    pub generation: Generation,
     pub services: Vec<ServiceZoneRequest>,
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq)]
+#[serde(rename_all = "snake_case")]
+pub enum ServiceEnsureStatus {
+    Updated,
+    NotUpdated,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq)]
+pub struct ServiceEnsureResponse {
+    // Returns the current generation number.
+    pub generation: Generation,
+    // Identifies whether or not this request updated the services on the sled.
+    pub status: ServiceEnsureStatus,
+}
```

Comment:
Why respond with the generation number at all? Under "normal operating conditions", it would be nice if "whoever made calls always knew the existing generation number". Well...

1. We didn't embed this generation number in the RSS handoff request...
2. ... so Nexus doesn't know what it is...
3. ... and the way RSS does setup is more complex than just "requesting all services at once", so it's not reliably Generation::new().next() -- it may have been called a couple times.

So, this is my workaround. Sled Agent simply tells the caller if it did an update or not, and also tells the caller "by the way, this is my current generation number".

Reply:
Follow-up option here -- we could just add a "nexus generation" as an optional to the thing stored in the ledger.

Should automatically be None, can be replaced by something more recent.